pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -770,7 +770,20 @@ func (m *kubeGenericRuntimeManager) killContainer(ctx context.Context, pod *v1.P
 	klog.V(2).InfoS("Killing container with a grace period", "pod", klog.KObj(pod), "podUID", pod.UID,
 		"containerName", containerName, "containerID", containerID.String(), "gracePeriod", gracePeriod)
 
-	err := m.runtimeService.StopContainer(ctx, containerID.ID, gracePeriod)
+	var err error
+	done := make(chan struct{})
```

Comment:
one more nit. I Think it's racy to have err set in the go func and out here. Maybe this should be a chan error, and send the error on it, then the condition below can be `case err = <-done:`

Reply:
ok,The test has passed.
Inside case, if !ok, then skip it, otherwise, if you get an err error, then return it.